modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/cell_based/medgun/medcells.dm

```diff
@@ -416,6 +437,37 @@
 	if(heals_left <= 0)
 		qdel(src)
 
+//Hardlight Roller Bed.
+/obj/structure/bed/roller/medigun
+	name = "Hardlight Roller Bed"
+	desc = "A Roller Bed made out of Hardlight"
+	icon = 'modular_skyrat/modules/modular_weapons/icons/obj/guns/mediguns/misc.dmi'
+	max_integrity = 1
+	buildstacktype = FALSE //It would not be good if people could use this to farm materials.
+	var/deploytime = 20 SECONDS //How long the roller beds lasts for without someone buckled to it.
+
+/obj/structure/bed/roller/medigun/Initialize()
+	. = ..()
+	addtimer(CALLBACK(src, .proc/CheckBed), deploytime)
+
+/obj/structure/bed/roller/medigun/proc/CheckBed()
+	if(!has_buckled_mobs())
+		qdel(src) //Deletes the roller bed, mostly meant to prevent stockpiling and clutter
+		return TRUE
```

Comment:
If it has buckled mobs, it doesn't return anything, which isn't really a good thing.

Also, use snake_case rather than UpperCamelCase, please.

Finally, if you could document that proc, that'd be nice. :)

Reply:
It is done.